core/templates/pages/contributor-dashboard-page/modal-templates/login-required-modal.component.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+/**
+ * @fileoverview Component for login required modal.
+ */
+
+@Component({
+  selector: 'login-required-modal',
+  templateUrl: './login-required-modal.component.html'
+})
+export class LoginRequiredModalContent {
+  constructor(public activeModal: NgbActiveModal) {}
```

Comment:
Why does `activeModal` have to be `public` here?

Reply:
I'm actually not entirely sure about this, but I got it from this NgbModal example:
https://ng-bootstrap.github.io/#/components/modal/examples#component

Reply:
Ah, I believe it's because the template HTML (core/templates/pages/contributor-dashboard-page/modal-templates/login-required-modal.component.html) will need access to the instance to call `close()`

Reply:
That makes sense, thanks!